lib/mapObjects/MiscObjects.cpp

```diff
@@ -359,6 +359,54 @@ std::vector<ObjectInstanceID> CGTeleport::getAllExits(bool excludeCurrent) const
 	return ret;
 }
 
+/**
+ * \brief Finds the next visible exit for a teleporter.
+ *
+ * This method tries to find the next visible exit for a teleporter based on the player color.
+ * If an exit is found, it returns the object instance. Otherwise, it returns a nullopt.
+ */
+std::optional<const CGObjectInstance*> CGTeleport::getNextVisibleExit(PlayerColor playerColor) const
+{
+	try {
+		std::vector<ObjectInstanceID> portals;
+
+		switch(type) {
+		case CGTeleport::ENTRANCE:
+		case CGTeleport::BOTH:
+			portals = getAllExits(false);
+			break;
+		case CGTeleport::EXIT:
+			portals = getAllEntrances(false);
+			break;
```

Comment:
What if portal have (for example) 1 exits but multiple entrances? Will player be able to navigate through all entrances?